### PR TITLE
feat(ugov): add uGOV token + tests + management

### PR DESCRIPTION
**Pull request by zapaz**
Creation of uGOV token + Tests + Token Management

### CODE REVIEW

File: contracts/UbiquityGovernance.sol [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+    ERC20Burnable,
+    ERC20Pausable
+{
+    UbiquityAlgorithmicDollarManager public manager;
+
+    // solhint-disable-next-line var-name-mixedcase
+    bytes32 public DOMAIN_SEPARATOR;
+    // keccak256("Permit(address owner,address spender,
+    //                   uint256 value,uint256 nonce,uint256 deadline)");
+    bytes32 public constant PERMIT_TYPEHASH =
```
Comment (zgorizzo69) [inline contracts/UbiquityGovernance.sol:20]: we should move that to the manager as it used both by ugov and uad

Comment (zapaz) [reply contracts/UbiquityGovernance.sol:20]: we can also make one preset like OZ does

make one ERC20Ubiquity preset with one interface IERC20Ubiquity (almost equal to actual IUbiquityAlgorithmicDollar.sol)

**Review by zgorizzo69 — COMMENTED** [review]
@zapaz what would be the pros of doing so ? 
Don't you think it is better for integration (meaning interaction from outside contracts) to have an interface than an interface implementation ?

**Comment by zgorizzo69** [comment]
> @zapaz what would be the pros of doing so ?
> Don't you think it is better for integration (meaning interaction from outside contracts) to have an interface than an interface implementation ?

File: contracts/UbiquityGovernance.sol (hunk 1 above)
Comment (zgorizzo69) [reply contracts/UbiquityGovernance.sol:20]: Actually there are so close that creating a preset make sense

Comment (zapaz) [reply contracts/UbiquityGovernance.sol:20]: done ERCUbiquity.sol

File: contracts/UbiquityGovernance.sol [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+        0x6e71edae12b1b97f4d1f60370fef10105fa2faae0126114a169c64845d6126c9;
+    mapping(address => uint256) public nonces;
+
+    // ----------- Modifiers -----------
+    modifier onlyMinter() {
+        require(
+            manager.hasRole(manager.UAD_MINTER_ROLE(), msg.sender),
```
[the remark this answers is not in the document]
Comment (zapaz) [reply contracts/UbiquityGovernance.sol:27]: unified as UBQ_XXXX_ROLE

File: contracts/UbiquityGovernance.sol [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+    /// @param account the account to burn from
+    /// @param amount the amount to burn
+    function burnFrom(address account, uint256 amount)
+        public
+        override(IUbiquityGovernance, ERC20Burnable)
+        onlyBurner
+        whenNotPaused
+    {
+        _burn(account, amount);
+        emit Burning(account, msg.sender, amount);
```
[the remark this answers is not in the document]
Comment (zapaz) [reply contracts/UbiquityGovernance.sol:142]: done

**Comment by zapaz** [comment]
dont't see the point : with last commit with have both ERCUbiquity.sol  implementation , and IERCUbiquity.sol interface (that is not that usefull)

**Comment by zgorizzo69** [comment]
good job @zapaz  :+1:

**Pull request merged**